pkg/networkservice/common/connect/translation/builder.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translation
+
+import "github.com/networkservicemesh/api/pkg/api/networkservice"
+
+// Builder is a builder for translation client
+type Builder struct {
+	requestOpts []RequestOption
+	connOpts    []ConnectionOption
+}
+
+// WithRequestOptions appends request options to the builder
+func (b *Builder) WithRequestOptions(requestOpts ...RequestOption) *Builder {
+	if b == nil {
+		b = new(Builder)
+	}
+
+	b.requestOpts = append(b.requestOpts, requestOpts...)
+
+	return b
+}
+
+// WithConnectionOptions appends connection options to the builder
+func (b *Builder) WithConnectionOptions(connOpts ...ConnectionOption) *Builder {
+	if b == nil {
+		b = new(Builder)
+	}
+
+	b.connOpts = append(b.connOpts, connOpts...)
+
+	return b
+}
+
+// Build returns a new translation client with set request, connection options
+func (b *Builder) Build() networkservice.NetworkServiceClient {
+	return &translationClient{
+		requestOpts: b.requestOpts,
+		connOpts:    b.connOpts,
+	}
+}
+
+// RequestOption is a request translation option
+type RequestOption func(request *networkservice.NetworkServiceRequest, clientConn *networkservice.Connection)
+
+// ReplaceMechanism replaces request.Connection.Mechanism with clientConn.Mechanism
+func ReplaceMechanism() RequestOption {
+	return func(request *networkservice.NetworkServiceRequest, clientConn *networkservice.Connection) {
+		request.GetConnection().Mechanism = clientConn.GetMechanism()
+	}
+}
+
+// ReplacePath replaces request.Connection.Path with clientConn.Path
+func ReplacePath() RequestOption {
```

Comment:
This option is not using.

Reply:
Now removed all options, translation client translates only mechanisms.